wit/request-reply.wit

```diff
@@ -0,0 +1,27 @@
+/// The request-reply interface allows a guest to send a message and await a response. This
+/// interface is considered optional as not all message services support the concept of
+/// request/reply. However, request/reply is a very common pattern in messaging and as such, we have
+/// included it as a core interface.
+interface request-reply {
+    use types.{client, message, error};
+
+    /// Options for a request/reply operation
+    record request-options {
```

Comment:
I'd suggest copying [`request-options` in wasi-http](https://github.com/WebAssembly/wasi-http/blob/main/wit/types.wit#L331) and making this a resource, with methods to set/get each field.  This has the advantage of allowing semver-compatible field additions over time.